version: '3'

services:

  dig_etl_engine:
    build: .
    volumes:
      - ./config_docker_sandbox.py:/app/dig-etl-engine/config.py
      - ./../mydig-projects:/projects_data
    ports:
      - "9999:9999"
    networks:
      - dig_net

  zookeeper:
    image: wurstmeister/zookeeper
    ports:
      - "2181:2181"
    networks:
      - dig_net

  kafka:
    image: wurstmeister/kafka
    ports:
      - "9092:9092"
    environment:
      KAFKA_ADVERTISED_HOST_NAME: kafka
      KAFKA_ZOOKEEPER_CONNECT: zookeeper:2181
      KAFKA_CREATE_TOPICS: "test:1:1"
    volumes:
#      - ./kafka/config/server.properties:/opt/kafka/config/server.properties
      - /var/run/docker.sock:/var/run/docker.sock
    depends_on:
      - zookeeper
    networks:
      - dig_net

#  logstash:
#    image: docker.elastic.co/logstash/logstash:5.5.1
#    environment:
#      PATH_CONFIG: /usr/share/logstash/pipeline
#    volumes:
#      - ./logstash/pipeline/logstash_sandbox.conf:/usr/share/logstash/pipeline/logstash.conf
#      - ./logstash/pipeline/:/usr/share/logstash/pipeline/
#      - ./logstash/settings/logstash.yml:/usr/share/logstash/config/logstash.yml
#    networks:
#      - dig_net
#    depends_on:
#      - kafka

networks:
  dig_net:
    driver: bridge
